drivers/net/tap/rte_eth_tap.c

```diff
@@ -1995,10 +2009,18 @@ eth_dev_tap_create(struct rte_vdev_device *vdev, const char *tap_name,
 	struct rte_eth_dev_data *data;
 	struct ifreq ifr;
 	int i;
-
+	int deviceNS = -1;
+	int originalNS = -1;
 	TAP_LOG(DEBUG, "%s device on numa %u", tuntap_name, rte_socket_id());
 
+	if (netns_use)
+	{
+		rte_ethdev_switch_netns(netns, &deviceNS, &originalNS);
+	}
+
 	dev = rte_eth_vdev_allocate(vdev, sizeof(*pmd));
+	dev->data->netns[0] = '\0';
```

Comment:
My guess is that the struct is already zeroed (See rte_eth_dev_data_alloc).
However, if you want to do it yourself again, don't just set the first byte to 0. set all the char array to 0

Reply:
Good point.